eventstore/store_consolidated_billable_events_test.go

```diff
@@ -0,0 +1,359 @@
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		err = db.Schema.Consolidate(eventio.EventFilter{
+			RangeStart: "2001-01-02",
+			RangeStop:  "2001-02-02",
+		})
+		Expect(err).To(MatchError(
+			MatchRegexp("violates check constraint"),
+		))
+
+		err = db.Schema.Consolidate(eventio.EventFilter{
+			RangeStart: "2001-01-01",
+			RangeStop:  "2001-07-01",
+		})
+		Expect(err).To(MatchError(
+			MatchRegexp("violates check constraint"),
+		))
+	})
+
+	It("Should fail if consolidate called twice for the same range", func() {
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		err = db.Schema.Consolidate(eventio.EventFilter{
+			RangeStart: "2001-01-01",
+			RangeStop:  "2001-02-01",
+		})
+		Expect(err).ToNot(HaveOccurred())
+		err = db.Schema.Consolidate(eventio.EventFilter{
+			RangeStart: "2001-01-01",
+			RangeStop:  "2001-02-01",
+		})
+		Expect(err).To(MatchError(MatchRegexp("duplicate key value violates unique constraint")))
+	})
+})
+
+var _ = Describe("IsRangeConsolidated", func() {
+	var (
+		cfg      eventstore.Config
+		scenario *testenv.TestScenario
+	)
+
+	BeforeEach(func() {
+		cfg = testenv.BasicConfig
+		scenario = testenv.NewTestScenario("2001-01-01T00:00")
+	})
+
+	It("Should return false if range has not been consolidated", func() {
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		result, err := db.Schema.IsRangeConsolidated(eventio.EventFilter{
+			RangeStart: "2001-01-01",
+			RangeStop:  "2001-02-01",
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(BeFalse())
+	})
+
+	It("Should return true if range has been consolidated", func() {
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		filter := eventio.EventFilter{
+			RangeStart: "2001-01-01",
+			RangeStop:  "2001-02-01",
+		}
+
+		db.Schema.Consolidate(filter)
+		result, err := db.Schema.IsRangeConsolidated(filter)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(BeTrue())
+	})
+
+})
+
+var _ = Describe("ConsolidateFullMonths", func() {
+	var (
+		cfg      eventstore.Config
+		scenario *testenv.TestScenario
+		now      string
+	)
+
+	BeforeEach(func() {
+		cfg = testenv.BasicConfig
+		scenario = testenv.NewTestScenario("2018-01-01T00:00")
+		now = time.Now().Format("2006-01-02")
+	})
+
+	It("Should not error when there are no time periods to consolidate", func() {
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		err = db.Schema.ConsolidateFullMonths("9001-01-01", now)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Should consolidate events which have not been consolidated yet ", func() {
+		db, err := scenario.Open(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		defer db.Close()
+
+		// Consolidate everything since 2001
+		consolidateSince := "2017-01-01"
```

Comment:
The comment doesn't match what this line is doing. Please update whichever is correct so they match.

Reply:
damn comments